lib/spack/spack/provider_index.py

```diff
@@ -67,17 +147,23 @@ def __init__(self, specs=None, restrict=False):
             self.update(spec)
 
     def update(self, spec):
+        """Update the provider index with the virtual specs provided by the
+        spec passed as input
```

Comment:
Love the addition of the docstring.

Nit: Is it more accurate and concise to say ``Update the provider index with additional virtual specs`` given the inclusion of the `Args` description?